activerecord/lib/active_record/table_metadata.rb

```diff
@@ -66,6 +66,10 @@ def polymorphic_association?
       association && association.polymorphic?
     end
 
+    def association_join_fk
+      association.send(:join_fk)
```

Comment:
We should avoid to use `send` if we want to call this method just make it public in the class. Also please rename it to `join_foreign_key`